src/algorand_applications/algorand_application_ids.rs

```diff
@@ -0,0 +1,86 @@
+use derive_more::{Constructor, Deref};
+
+use crate::{
+    algorand_address::AlgorandAddress,
+    algorand_applications::algorand_application_args::AlgorandApplicationArg,
+    algorand_traits::ToApplicationArg,
+    algorand_types::{Byte, Bytes, Result},
+    crypto_utils::sha512_256_hash_bytes,
+};
+
+#[derive(Clone, Debug, Eq, PartialEq, Default, Constructor, Deref)]
```

Comment:
Just a curiosity...
Is there a standard subset you choose to derive from?

Reply:
Sort of. It's always useful to derive `Debug` because that means you can print out a struct like so `println!("{:?}", my_struct);` which is useful for...debugging :p.

`Clone` is almost always useful too for obvious reasons.

`Eq` & `PartialEq` are most often useful for tests, so you can at least compare the things (via `assert_eq` for eg).

`Default` is also always useful in tests for quickly instantiating something. 

As for `Constructor` and `Deref`, they're not in the rust std lib, as you can see from them being important. But `Constructor` is a super simple way of lazily defining `MyStruct::new` when you want to pass in exactly the types in exactly the order they appear in the struct for. Which is not _always_ the case but quite often is.

Finally, `Deref` is shear laziness by me, so I don't have to use `my_struct.0` to access the inner tuple. It's more useful in the case a tuple struct of a vec, eg `struct MyStruct(Vec<T>)`, because with `Deref`, and can treat `MyStruct` exactly as I would the underlying `Vec`, ie, I can call all the underlying vec methods straight on the struct: `my_struct.len()`, `my_struct.iter()...` etc..